README.md

```diff
@@ -39,26 +55,70 @@ const options = JSON.parse(process.env.REACT_APP_FIREBASE_CONFIG!)
 
-```
-playwrightFirebasePlugin(serviceAccount, options, uid, base, version)
-```
+```ts
+//example.spec.ts
+import { expect } from '@playwright/test'
+import { test } from '../auth.setup' // <----- here we import test from our auth.setup.ts.
+import { Page } from '@playwright/test'
 
-Where your secrets are stored in a `.env` file. Make sure to **NOT COMMIT THIS FILE**.
-Now, by using the new `test` we can incorporate the `auth` generated from the package.
+// We now access auth in exactly the same method as we do page.
+test('has title', async ({ page, auth }: { page: Page; auth: any }) => {
+  await page.goto('/', { waitUntil: 'networkidle' })
+  await auth.login(page) // <-- we need to pass in the page object here.
 
-```
-import { expect } from '@playwright/test';
-import { test } from '../auth.setup'
-import { Page } from '@playwright/test';
-test('has title', async ({ page, auth }: { page: Page, auth: any }) => {
-  await page.goto('/', { waitUntil: 'networkidle' });
-  await auth.login(page)
-
-  const txt = page.getByText('Welcome! You are now logged in')
+  const txt = await page.getByText('Welcome! You are now logged in')
   await expect(txt).toBeVisible()
+
   await auth.logout(page)
+
   await expect(txt).not.toBeVisible()
-});
+})
 ```
+
+It's recommended to use `await` for your `expect` assertions after logging in/out, as the Firebase authentication is likely tied to states that require re-rendering of your application.
+
+### TypeScript
+
+If you're using Typescript, one small addition you'll need to make is to add the type `Credentials` to your `playwright.config.ts` such that
+
+```ts
+import { Credentials } from '@nearform/playwright-firebase'
+
+export default defineConfig<Credentials>({
+  ...
+})
+```
+
+<a name="motivation"></a>
+
+## Motivation
+
+This package is built as a plugin for Playwright testing for the use-case of Firebase authentication. There are two methods of automating a login procedure in Playwright tests:
+
+1. As a normal user would: inserting a real username and password into the intended fields.
+2. Authenticating via the Firebase SDK directly
+
+This plugin was developed with the 2nd method in mind as it is
+
+- Provider agnostic: Does not need to know the specifics of the authentication provider
+- A faster way of logging in, so you can focus on testing
+- Better security than using a username and password.
+
+<a name="example"></a>
+
+## Example
+
+Within this repo we have an `example/` folder that contains a sample React application for authenticating with the Google Provider. You'll need to setup the Firebase environment variables as described above in the setup section, but the rest is taken care of.
+
+1. Clone this repository
+2. `cd ./example`
+3. `npm i`
+4. `npm init playwright@latest`
```

Comment:
why is this needed? doesn't the example already contain an "initialized" version of playwright?